pkg/pipeline/write/write_loki_test.go

```diff
@@ -285,3 +291,57 @@ parameters:
 		require.Fail(t, "timeout while waiting for LokiWriter to forward data")
 	}
 }
+
+func buildFlow(t time.Time) config.GenericMap {
+	return config.GenericMap{
+		"timestamp": float64(t.UnixMilli()),
+		"srcIP":     "10.0.0." + strconv.Itoa(rand.Intn(20)),
+		"dstIP":     "10.0.0." + strconv.Itoa(rand.Intn(20)),
+		"flags":     "SYN",
+		"bytes":     rand.Intn(100),
+		"packets":   rand.Intn(10),
+		"latency":   rand.Float64(),
+	}
+}
+
+func hundredFlows() []config.GenericMap {
+	flows := make([]config.GenericMap, 100)
+	t := time.Date(2022, time.August, 31, 8, 0, 0, 0, time.Local)
+	for i := 0; i < 100; i++ {
+		t = t.Add(time.Second)
+		flows[i] = buildFlow(t)
+	}
+	return flows
+}
+
+func BenchmarkWriteLoki(b *testing.B) {
+	log.SetLevel(log.ErrorLevel)
+	lokiFlows := make(chan map[string]interface{}, 256)
+	fakeLoki := httptest.NewServer(test.FakeLokiHandler(lokiFlows))
+
+	params := api.WriteLoki{
+		URL: fakeLoki.URL,
+		StaticLabels: model.LabelSet{
+			"app": "flp-benchmark",
+		},
+		Labels:         []string{"srcIP", "dstIP"},
+		TimestampLabel: "timestamp",
+		TimestampScale: "1ms",
+	}
+
+	loki, err := NewWriteLoki(config.StageParam{Write: &config.Write{Loki: &params}})
+	require.NoError(b, err)
+
+	for i := 0; i < b.N; i++ {
+		loki.Write(hundredFlows())
+	}
+
+	i := 0
+	for range lokiFlows {
+		i++
+		if i == 100*b.N {
+			close(lokiFlows)
+			break
+		}
+	}
```

Comment:
What's the purpose of this loop?

Reply:
Reading on the channel is necessary to not block the flows processing. I haven't looked much further in `test.FakeLokiHandler`, I'd agree this code is not super elegant maybe something better can be done, i'll look more into it